flink-runtime/src/main/java/org/apache/flink/runtime/dispatcher/Dispatcher.java

```
@@ -404,7 +406,29 @@ public void start() throws Exception {
 		final JobManagerRunner jobManagerRunner = jobManagerRunners.get(jobId);
 
 		if (jobManagerRunner != null) {
-			return jobManagerRunner.getJobManagerGateway().requestJobStatus(timeout);
+				CompletableFuture<JobStatus> statusFuture = jobManagerRunner.getJobManagerGateway().requestJobStatus(timeout);
+				statusFuture.handle((JobStatus status, Throwable throwable) -> {
+					if (throwable != null) {
+						Throwable error = ExceptionUtils.stripCompletionException(throwable);
+
+						if (error instanceof FencingTokenException) {
```

missing else block causes other exceptions to be swallowed...